### DISCUSSION TITLE
[ANN] IKOS 3.2-rc1 release candidate

### DISCUSSION THREAD

**Discussion by ivanperez-keera** [NASA-SW-VnV/ikos#253] [Announcements]
Hi everyone!

**EDIT (2024-01-21): The new version of IKOS, 3.2, has now been released. Please ignore this discussion and follow the instructions in https://github.com/NASA-SW-VnV/ikos#installation. Thanks!**

We've just made available a first release candidate for IKOS 3.2.

The tag is https://github.com/NASA-SW-VnV/ikos/releases/tag/v3.2-rc1.

We kindly ask everyone to test it and let us know if you find any issues we should fix before the final release.

## Changes

A list of the issues addressed since the prior release can be seen here:

https://github.com/NASA-SW-VnV/ikos/milestone/1?closed=1

A more detailed list of changes is available at:

https://github.com/NASA-SW-VnV/ikos/compare/v3.1...v3.2-rc1

## Installation

You can clone this new version directly from the repo using the tag v3.2-rc1.

To install this release candidate with homebrew, use the following fork:

```
$ brew install ivanperez-keera/homebrew-core/ikos
```

Note: The url to be used with homebrew once the release is final will continue to be `NASA-SW-VnV/core/ikos` as usual.

The following comment may help install IKOS on a fresh new machine or in docker using homebrew:

https://github.com/NASA-SW-VnV/ikos/issues/225#issuecomment-1859070104

## Further notes

Please, open issues for any problems you find during installation, or anything else that you think needs to be fixed.

Thanks everyone for being a part of the IKOS community!

**Comment by ivanperez-keera** [comment]
All installation issues (https://github.com/NASA-SW-VnV/homebrew-core/issues/9) seem solved, so I think we are good to proceed.

This issue will be closed when I merge #256, the fix to #255.

The homebrew formula will be merged right after (pointing to the correct version 3.2 of IKOS, instead of 3.2-rc1).

I'll give it a day or so for anyone to come forward with any issues they have. Otherwise, I'll go ahead and release.

**Comment by ivanperez-keera** [comment]
The new version has been released.

**Comment by CIPop** [comment]
This doesn't work for me:

```
brew install ivanperez-keera/homebrew-core/ikos
Error: Formulae found in multiple taps:
       * ivanperez-keera/core/apron
       * nasa-sw-vnv/core/apron

Please use the fully-qualified name (e.g. ivanperez-keera/core/apron) to refer to the formula.
```

**Reply by CIPop** [reply]
I tried to:
`brew untap nasa-sw-vnv/core`

But then it still fails:

```
brew install ivanperez-keera/homebrew-core/ikos
==> Fetching ivanperez-keera/core/ikos
==> Downloading https://files.pythonhosted.org/packages/7e/ae/26808275fc76bf2832deb10d3a3ed3107bc4de01b85dcccbe525f2cd6d1e/Pygments-2.4.2.tar.gz
Already downloaded: /home/cristian/.cache/Homebrew/downloads/6ca94f739e0ce8b61064d8fa84dcd6adf34e71e1a606f1ec9403a187965de88c--Pygments-2.4.2.tar.gz
==> Downloading https://github.com/NASA-SW-VnV/ikos/archive/refs/tags/v3.2.tar.gz
Already downloaded: /home/cristian/.cache/Homebrew/downloads/1fe6d9981957abde286f236e44082fb056f93bd4820392065c8fdb673e17dd1d--ikos-3.2.tar.gz
==> Installing ikos from ivanperez-keera/core
==> python3 -m venv --system-site-packages --without-pip /home/linuxbrew/.linuxbrew/Cellar/ikos/3.2_1/libexec
==> python3 -m pip --python=/home/linuxbrew/.linuxbrew/Cellar/ikos/3.2_1/libexec/bin/python install /tmp/ikos--Pygments-20240121-17824-m2e18b/Pygments-2.4.2
==> cmake -G Unix Makefiles -DGMP_ROOT=/home/linuxbrew/.linuxbrew/opt/gmp -DMPFR_ROOT=/home/linuxbrew/.linuxbrew/opt/mpfr -DPPL_ROOT=/home/linuxbrew/.linuxbrew/opt/ppl -DAPRON_ROOT=/home/linuxbrew/.linuxbrew/opt/apron -DCUSTOM_BOO
==> make install
Last 15 lines from /home/cristian/.cache/Homebrew/Logs/ikos/04.make:
  892 |     if (is_zero_bit(p, n)) {
      |         ~~~~~~~~~~~^~~~~~
[ 16%] Building CXX object ar/CMakeFiles/ikos-ar.dir/src/semantic/data_layout.cpp.o
make[2]: *** [ar/CMakeFiles/ikos-ar.dir/build.make:188: ar/CMakeFiles/ikos-ar.dir/src/pass/simplify_upcast_comparison.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 18%] Building CXX object frontend/llvm/CMakeFiles/ikos-pp-lib.dir/src/pass/remove_printf_calls.cpp.o
[ 18%] Building CXX object frontend/llvm/CMakeFiles/ikos-pp-lib.dir/src/pass/remove_unreachable_blocks.cpp.o
[ 18%] Linking CXX static library libikos-pp.a
[ 18%] Built target ikos-pp-lib
[ 18%] Building CXX object frontend/llvm/CMakeFiles/ikos-pp.dir/src/ikos_pp.cpp.o
make[1]: *** [CMakeFiles/Makefile2:1528: ar/CMakeFiles/ikos-ar.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 19%] Linking CXX executable ikos-pp
[ 19%] Built target ikos-pp
make: *** [Makefile:146: all] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
  https://github.com/ivanperez-keera/homebrew-core/issues
```

**Reply by ivanperez-keera** [reply]
Now that the release is final, you should remove anything related to `ivanperez-keera/homebrew-core` and just install normally from `nasa-sw-vnv/core/ikos` following the instructions in https://github.com/NASA-SW-VnV/ikos#installation
